### PR TITLE
my OP_RETURN stuff

### PR DESCRIPTION
**Pull request by lorenz10**

### CODE REVIEW

**Comment by bananenwilly** [comment]
`Built test:test.
00:03 +157 -1: test/integration/transactions_test.dart: bitcoinjs-lib (transactions) can create an "null data" Transaction [E]                                                                                                                                                                             
  Expected: '01000000019d344070eac3fe6e394a16d06d7704a7d5c0a10eb2a2c16bc98842b7cc20d561000000006a47304402200852194e22d2b5faf9db66407d769b13278708b77e55df5d9c8638367af4c0870220638083bdaf06d8147ad4bfaddb975a2a4a056cca806a717e956d334c01482b3a0121029f50f51d63b345039a290c94bffd3180c99ed659ff6ea6b1242bca47eb93b59fffffffff0100000000000000002e6a2c486579207468697320697320612072616e646f6d20737472696e6720776974686f757420426974636f696e7300000000'
    Actual: '01000000019d344070eac3fe6e394a16d06d7704a7d5c0a10eb2a2c16bc98842b7cc20d561000000006b483045022100cf9cdd4cdc7c20ec52ddd2d9a944a9c2e36d4ad72be94e2cbc3cde092ac5a9e002204badff06f7439b7f82f1adcfbdbb4091d11c2e478a8d4b6834b37a41ed4f215d0121029f50f51d63b345039a290c94bffd3180c99ed659ff6ea6b1242bca47eb93b59fffffffff0100000000000000002b6a29486579207468697320697320612072616e646f6d20737472696e6720776974686f757420636f696e7300000000'
     Which: is different.
            Expected: ... 1000000006a473044022 ...
              Actual: ... 1000000006b483045022 ...
                                    ^
             Differ at offset 83
  
  package:test_api                              expect
  test/integration/transactions_test.dart 77:7  main.<fn>.<fn>
  
00:04 +157 -1: loading test/integration/transactions_test.dart                                                                                                                                                                                                                                             
Consider enabling the flag chain-stack-traces to receive more detailed exceptions.
For example, 'dart test --chain-stack-traces'.
00:05 +283 -1: Some tests failed.             `

File: lib/src/transaction_builder.dart [hunk 1]
```diff
@@ -10,6 +12,8 @@ import 'payments/p2pkh.dart';
 import 'payments/p2wpkh.dart';
 import 'classify.dart';
 
+const int MAX_OP_RETURN_SIZE = 256;
```
Comment (bananenwilly) [inline lib/src/transaction_builder.dart:15]: This should rather be part of the NetworkType model as different coins have different OP_RETURN sizes.

**Comment by lorenz10** [comment]
> Built test:test.
> 00:03 +157 -1: test/integration/transactions_test.dart: bitcoinjs-lib (transactions) can create an "null data" Transaction [E]                                                                                                                                                                             
>   Expected: '01000000019d344070eac3fe6e394a16d06d7704a7d5c0a10eb2a2c16bc98842b7cc20d561000000006a47304402200852194e22d2b5faf9db66407d769b13278708b77e55df5d9c8638367af4c0870220638083bdaf06d8147ad4bfaddb975a2a4a056cca806a717e956d334c01482b3a0121029f50f51d63b345039a290c94bffd3180c99ed659ff6ea6b1242bca47eb93b59fffffffff0100000000000000002e6a2c486579207468697320697320612072616e646f6d20737472696e6720776974686f757420426974636f696e7300000000'
>     Actual: '01000000019d344070eac3fe6e394a16d06d7704a7d5c0a10eb2a2c16bc98842b7cc20d561000000006b483045022100cf9cdd4cdc7c20ec52ddd2d9a944a9c2e36d4ad72be94e2cbc3cde092ac5a9e002204badff06f7439b7f82f1adcfbdbb4091d11c2e478a8d4b6834b37a41ed4f215d0121029f50f51d63b345039a290c94bffd3180c99ed659ff6ea6b1242bca47eb93b59fffffffff0100000000000000002b6a29486579207468697320697320612072616e646f6d20737472696e6720776974686f757420636f696e7300000000'
>      Which: is different.
>             Expected: ... 1000000006a473044022 ...
>               Actual: ... 1000000006b483045022 ...
>                                     ^
>              Differ at offset 83
>   
>   package:test_api                              expect
>   test/integration/transactions_test.dart 77:7  main.<fn>.<fn>
>   
> 00:04 +157 -1: loading test/integration/transactions_test.dart                                                                                                                                                                                                                                             
> Consider enabling the flag chain-stack-traces to receive more detailed exceptions.
> For example, 'dart test --chain-stack-traces'.
> 00:05 +283 -1: Some tests failed.

I'm struggling to solve this one.

File: lib/src/transaction_builder.dart (hunk 1 above)
Comment (lorenz10) [reply lib/src/transaction_builder.dart]: fixed

File: lib/src/models/networks.dart [hunk 2]
```diff
@@ -38,12 +40,14 @@ final bitcoin = new NetworkType(
...
 
 final testnet = new NetworkType(
     messagePrefix: '\x18Bitcoin Signed Message:\n',
     bech32: 'tb',
     bip32: new Bip32Type(public: 0x043587cf, private: 0x04358394),
     pubKeyHash: 0x6f,
     scriptHash: 0xc4,
-    wif: 0xef);
+    wif: 0xef,
+    opreturnSize: 256);
```
Comment (bananenwilly) [inline lib/src/models/networks.dart:53]: https://stackoverflow.com/questions/24845429/maximum-size-of-data-bitcoin-op-return-tx-can-handle

It's max 80 for BTC, peercoin has a pretty big one with 256

File: lib/src/models/networks.dart [hunk 3]
```diff
@@ -38,12 +40,14 @@ final bitcoin = new NetworkType(
     bip32: new Bip32Type(public: 0x0488b21e, private: 0x0488ade4),
     pubKeyHash: 0x00,
     scriptHash: 0x05,
-    wif: 0x80);
+    wif: 0x80,
+    opreturnSize: 256);
```
Comment (bananenwilly) [inline lib/src/models/networks.dart:45]: 80

**Comment by peerchemist** [comment]
Lets port this test to Peercoin-testnet so we can check out what is wrong easier.

**Comment by bananenwilly** [comment]
can confirm that this is working, just the tests are broken

https://chainz.cryptoid.info/ppc-test/tx.dws?884170.htm

**Comment by peerchemist** [comment]
Can we get going with this?

File: lib/src/models/networks.dart (hunk 2 above)
Comment (lorenz10) [reply lib/src/models/networks.dart]: fixed

File: lib/src/models/networks.dart (hunk 3 above)
Comment (lorenz10) [reply lib/src/models/networks.dart:45]: fixed

**Pull request merged**